tools/c7n_gcp/c7n_gcp/policy.py

```diff
@@ -101,8 +103,47 @@ def run(self, event, context):
         return PullMode.run(self)
 
 
+class FunctionModeImpl(FunctionMode):
+    def resolve_resources(self, event):
+        raise NotImplementedError("subclass responsibility")  # pragma: no cover
+
+    def run(self, event, context):
+        """Execute a gcp serverless model"""
+        from c7n.actions import EventAction
+
+        s = time.time()
+        resources = self.resolve_resources(event)
+        if not resources:
+            return  # pragma: no cover
+
+        resources = self.policy.resource_manager.filter_resources(resources, event)
+
+        if not resources:
+            self.policy.log.info(
+                "policy: %s resources: %s no resources found"
+                % (self.policy.name, self.policy.resource_type)
+            )
+            return
+        rt = time.time() - s
+
+        with self.policy.ctx as ctx:
```

Comment:
Similar to the azure change, but different, the context wasn't being entered at all, which means that the output wasn't being written at all for executions, nor metrics output. This change uses the context, which triggers the metrics and output saving, and also writes out the expected `resources.json` file.